enigma/src/main/java/cuchaz/enigma/translation/mapping/EntryMapping.java

```diff
@@ -3,47 +3,25 @@
 import javax.annotation.Nonnull;
 import javax.annotation.Nullable;
 
-public class EntryMapping {
-	private final String targetName;
-	private final AccessModifier accessModifier;
-	private final @Nullable String javadoc;
-
-	public EntryMapping(@Nonnull String targetName) {
+public record EntryMapping(
+		@Nullable String targetName,
+		@Nonnull AccessModifier accessModifier,
+		@Nullable String javadoc
+) {
+	public static final EntryMapping DEFAULT = new EntryMapping(null, AccessModifier.UNCHANGED, null);
+
+	public EntryMapping(@Nullable String targetName) {
 		this(targetName, AccessModifier.UNCHANGED);
 	}
 
-	public EntryMapping(@Nonnull String targetName, @Nullable String javadoc) {
+	public EntryMapping(@Nullable String targetName, @Nullable String javadoc) {
 		this(targetName, AccessModifier.UNCHANGED, javadoc);
 	}
 
-	public EntryMapping(@Nonnull String targetName, AccessModifier accessModifier) {
+	public EntryMapping(@Nullable String targetName, AccessModifier accessModifier) {
 		this(targetName, accessModifier, null);
 	}
 
```

Comment:
Consider adding the canonical constructor to ensure `accessModifier` is not `null`

```java
public EntryMapping {
	if (accessModifier == null) {
		accessModifier = AccessModifier.UNCHANGED;
	}
}
```

Reply:
Oh, that's some neat syntax! And good point, let me add an error message too though. I did test and fixed any occurrences of this getting passed null (there was only one from what I've tested and it broke loading mappings so I noticed immediately haha), but just in case.